File: admin-panel/foodagram-admin-panel/src/models/content/dto/CommentResponseDto.ts
export interface CommentResponseDto {
    id: string;
    postId: string;
    userId: string;
    content: string;
    createdByUsername: string;
    parentReplyId?: string | null;
    replies: CommentResponseDto[];
    upvoteCount: number;
    downvoteCount: number;
    isDeleted: boolean;
    edited: boolean;
    createdAt: string; // ISO string (e.g., "2025-04-09T11:47:58.124Z")
    updatedAt: string;
}

File: api/clients/src/main/java/com/foodagram/clients/content/dto/CommentResponseDto.java
package com.foodagram.clients.content.dto;

import com.foodagram.clients.content.enums.PostType;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class CommentResponseDto {
    private UUID id;
    private UUID postId;
    private UUID userId;
    private String content;
    private String createdByUsername;
    private UUID parentReplyId;
    private List<CommentResponseDto> replies;
    private Long upvoteCount;
    private Long downvoteCount;
    private boolean isDeleted;
    private boolean edited;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}
